src/Illuminate/Support/Arr.php

```diff
@@ -235,14 +235,15 @@ public static function get($array, $key, $default = null)
 
 		if (isset($array[$key])) return $array[$key];
 
-		foreach (explode('.', $key) as $segment)
+		$key = strtok($key, '.');
+
+		while ($key !== false)
 		{
-			if ( ! is_array($array) || ! array_key_exists($segment, $array))
-			{
-				return value($default);
-			}
+			if ( ! isset($array[$key])) return value($default);
```

Comment:
@barryvdh When I put back the `if ( ! is_array($array) || ! array_key_exists($segment, $array))` the speed improvement is gone.

So it would appear that the optimization is this line instead of using strtok.

Nevertheless, I still think the `!is_array($array)` is not needed at all and the `!isset()` is a better check.